integration_test/third_party_apps_test.go

```diff
@@ -187,6 +190,34 @@ type expectedEntry struct {
 	FieldMatchers map[string]string `yaml:"field_matchers"`
 }
 
+// expectedMetricsEntry encodes a series of assertions about what data we expect
+// to see in the metrics backend.
+type expectedMetricEntry struct {
+	// The metric type, for example workload.googleapis.com/apache.current_connections.
+	Type string `yaml:"type"`
+	// The value type, for example INT64.
+	ValueType string `yaml:"value_type"`
+	// The kind, for example GAUGE.
+	Kind string `yaml:"kind"`
+	// The monitored resource, for example gce_instance.
+	// Currently we only test with gce_instance, so we expect
+	// all expectedMetricsEntries to have gce_instance.
+	MonitoredResource string `yaml:"monitored_resource"`
+	// Mapping of expected label keys to value patterns.
+	// Patterns are RE2 regular expressions.
+	Labels map[string]string `yaml:"labels"`
+	// If set to true, skip this entry during the test.
```

Comment:
Why not try to retrieve it, but not fail the test if it's not there? That way we can still validate it if it's present…

Reply:
Are you referring to `labels` or `optional`? The highlighted part includes bits of both.

Reply:
I was referring specifically to the "skip this entry during the test" in line 209.

Reply:
Updated behaviour & wording.